tensorflow/compiler/xla/literal_util_test.cc

```diff
@@ -730,6 +766,38 @@ TEST_F(LiteralUtilTest, CopyScalars) {
   EXPECT_EQ(LiteralUtil::Get<uint32>(*vect, {4}), 17);
 }
 
+TEST_F(LiteralUtilTest, F16) {
+  auto m1 = LiteralUtil::CreateFromShape(ShapeUtil::MakeShape(F16, {2, 2}));
+  Literal* l1 = m1.get();
+  const char* d1 = (const char*)LiteralUtil::InternalData(*l1);
+  EXPECT_EQ(d1[0], 0);
+  EXPECT_EQ(d1[1], 0);
+  EXPECT_EQ(d1[2], 0);
+  EXPECT_EQ(d1[3], 0);
+  EXPECT_EQ(d1[4], 0);
+  EXPECT_EQ(d1[5], 0);
+  EXPECT_EQ(d1[6], 0);
+  EXPECT_EQ(d1[7], 0);
+  EXPECT_EQ(LiteralUtil::InternalData(*l1),
+            LiteralUtil::MutableInternalData(l1));
+
+  half h1 = (half)1.0f;
+  half h2 = (half)2.0f;
+  auto m2 = LiteralUtil::CreateR2<half>({{h1, h2}, {h2, h1}});
+  Literal* l2 = m2.get();
+  const char* d2 = (const char*)LiteralUtil::InternalData(*l2);
+  EXPECT_EQ(d2[0], 0);
```

Comment:
Just a check that getting data via the InternalData and MutableInternalData work as expected.    It does veryify the endianess, but also that there are no weird things going on with 'bytes'.  I don't trust the protobuf 'bytes' as it is implemented as a std::string, which seems like a weird container to choose when std::vector<uint8> would work.